debounce search, move to twins view

karladler opened the pull request.

karladler committed:
debounce search, move to twins view

datrinh commented on dapps/assets/src/components/DataContainer.ts:
```diff
@@ -48,14 +48,22 @@ export default class DataContainerComponent extends mixins(EvanComponent) {
     this.page = 0;
     this.data = [];
-    const { result, total } = await this.search.query(this.type, { searchTerm: this.searchTerm });
+    this.searchTerm = searchTerm;
```
Is this side effect necessary?
Before ```this.searchTerm = to.params.query;``` is called

karladler replied:
yes, bc we call this always when the searchTerm changed, not only once after mounting

datrinh commented on dapps/assets/src/components/digitaltwins/DigitalTwins.scss:
```diff
@@ -0,0 +1,37 @@
+  }
+}
+
+/deep/ .evan-table {
```
evan-table component is available now. These styles can be removed.

datrinh commented on dapps/assets/src/components/digitaltwins/DigitalTwins.ts:
```diff
@@ -48,6 +50,15 @@ export default class DigitalTwinsComponent extends mixins(EvanComponent) {
+  }) search: Function;
+
+  @Watch('searchTerm')
+    onSearchtermChanged(searchTerm: string, oldVal: string) {
```
```suggestion
    onSearchTermChanged(searchTerm: string, oldVal: string) {
```
Fix function name accordingly

datrinh commented on dapps/assets/src/components/digitaltwins/DigitalTwins.ts:
```diff
@@ -66,5 +77,17 @@ export default class DigitalTwinsComponent extends mixins(EvanComponent) {
+    }
+  }
+
+  searchHandlerDebounced = debounce(
```
Why not debounce in searchHandler and switch depending on debounce option parameter?

karladler committed:
improve search UX

- [CORE-744] improve search UX by intercepting cmd-f functionality with our search

karladler committed:
Merge branch 'refs/heads/develop' into feature/CORE-744-search-review-changes

# Conflicts:
#	dapps/assets/src/components/Assets.ts
#	dapps/assets/src/components/digitaltwins/DigitalTwins.vue

karladler committed:
fix double search requests & PR fixes
